src/main/java/hudson/plugins/copyartifact/Copier.java

```diff
@@ -34,8 +39,34 @@
      * @param srcDir Source for upcoming file copy
      * @param baseTargetDir Base target dir for upcoming file copy (the copy-artifact
      *   build step may later specify a deeper target dir)
+     *
+     * @since TODO ?
      */
-    public abstract void init(Run src, AbstractBuild<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException;
+    public void init(Run src, Run<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException {
```

Comment:
@jglick interesting one here if I introduce the generic types on `Run src` ... it results in a compile time error re ambiguous method signatures. I need to read the Java language spec now lol